example/v2_framework.cpp

```diff
@@ -407,6 +419,23 @@ static_assert(is_of_type<joule / watt, derived_unit<struct joule, per<struct wat
 }  // namespace units::isq::si
 
 
+template<typename T, Dimension auto D, Unit auto U>
+inline constexpr bool is_exactly_quantity_of =
+  is_same_v<decltype(T::dimension), decltype(D)> && is_same_v<decltype(T::unit), decltype(U)>;
+
+namespace units::isq::si {
+
+// quantity tests
+static_assert(
+  is_exactly_quantity_of<decltype(4 * length[km] / (2 * length[m])), one_dim, derived_unit<kilometre, per<metre>>>);
```

Comment:
Why isn't this simplified to `derived_unit<kilo<one>>`? I know the SI brochure says
> Prefix symbols can neither stand alone nor be attached to the number 1, the symbol for the
unit one. Similarly, prefix names cannot be attached to the name of the unit one, that is, to
the word “one”.

However, I think it's fine as an artifact of programming. It should certainly simplify the implementation, and should be fine as long as the interface to the real world, like the printing operations, are compliant.

It also looks more like the actual result, which looks like `$4 km / 2 m = 2 × 10³$`. And something like `derived_unit<kilometre, per<megametre>>` becomes hard to parse, whereas `derived_unit<one, per<kilo<one>>>` is more clear.

Reply:
Of course, the resulting unit is equivalent to the unit one, which should have no symbol or prefix.

> I think it's fine as an artifact of programming

It is not an artifact of programming. We can easily simplify this to be always `one`. However, don't you think that allowing a user to define a dimensionless quantity that may be printed as `123 m/m` or similar might be a really common case?

I mean that the following could be supported:

```cpp
auto v1 = 4 * length[km] / (2 * length[m]);  // km/m
auto v2 = 4 * length[m] / (2 * length[m]);  // m/m
auto v3 = quantity_cast<one>(v2);  // one
auto v4 = 123 * dimensionless[m/m]; // m/m
```

Is that something that we should allow, or should we always force a simplification on scaled units?

Reply:
As you see in the code I think that dimensions are mostly done, and now I am trying to research different options for unit handling and then choose the best solution.

In the end, we will have to decide about implicit/explicit conversions and comparability of such "strange" quantities ;-)

Reply:
> [...] might be a really common case?

Isn't it more common to divide two quantities of the same dimension and expect an actual unit-less number?

> Is that something that we should allow, or should we always force a simplification on scaled units?

If you compare this to other special handling of dimensionless, like in quantity construction, this one has more dubious benefits. I'd lean toward always simplifying. `v4` shows that it's already allowed somewhat.

Reply:
> Isn't it more common to divide two quantities of the same dimension and expect an actual unit-less number?

Isn't it the same as the fact that from `10 * energy[J] / (2 * time[s])` I will get `5 * <energy/time, joule/second>` rather than `5 * <power, watt>`?

Reply:
> If you compare this to other special handling of dimensionless, like in quantity construction

Quantity construction will not be affected as the result of `4 * length[km] / (2 * length[m]);` is `2 * <one, km/m>` which is still a dimensionless quantity.

Reply:
I do not yet know what the best solution is, but I want to try different approaches, see which one can actually be implemented, and then find the best solution from those.

Of course we can eliminate some scenarios right away if we decide that they are confusing or simply wrong.

Reply:
> > Isn't it more common to divide two quantities of the same dimension and expect an actual unit-less number?
> 
> Isn't it the same as the fact that from `10 * energy[J] / (2 * time[s])` I will get `5 * <energy/time, joule/second>` rather than `5 * <power, watt>`?

It makes sense when you look at it that way. Will dimensional analysis ever simplify it? Or will the (dimension, unit) pair of `(1 * length[km]) / (1 * length[m]) * (1 * time[s])` be (time, `kilometre/metre * second`)?

Reply:
This is a very good question. Do you have any recommendations/ideas what is the best way or at least what we should consider? I have some but would like to hear yours as well.

Reply:
BTW, please note that tomorrow all day I will be on the plane flying to CppCon. Among others, I will host the discussion panel on Tuesday, where I will share our ideas about the V2 framework and will try to get early feedback, ideas, and requirements.

Reply:
Also, please note that in the new design, we cannot print a unit in the base symbols of the specific system because at least for now we do not have such knowledge (previously it was embedded in the dimensions).

Reply:
The simplest would be to traffic in base units. Named operations can be provided for more special cases.

For example, `divide(kilometer, meter)` equals the unit of "kilometer per meter", and `kilometer / meter` equals `simplify(divide(kilometer, meter))` equals `1000`.

These are only two extremes, though. There could be an arbitrary number of steps required to convert an equation as written to its simplest form.